core/lib/tasks/solidus/split_promotions_with_any_match_policy.rake

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+namespace :solidus do
+  desc "Split Promotions with 'any' match policy"
+  task split_promotions_with_any_match_policy: :environment do
+    Spree::Promotion.where(match_policy: :any).includes(:promotion_rules).all.each do |promotion|
+      if promotion.promotion_rules.length <= 1
+        promotion.update!(match_policy: :all)
+      elsif promotion.active?
+        promotion_rules.map do |rule|
+          new_promotion = promotion.dup
+          new_promotion.rules = rule
```

Comment:
I'm not sure this actually works, shouldn't it be `new_promotion.rules << rule`?

Also, given there's quite some logic here, I think it would be worth adding a spec for this task, so we can confirm it works at least at a basic level.

Reply:
I've added a spec, and you were right, there were a few bugs in there